stubs/requests/requests/sessions.pyi

```diff
@@ -48,12 +48,16 @@ class SessionRedirectMixin:
 _Data: TypeAlias = str | bytes | Mapping[str, Any] | Iterable[tuple[str, str | None]] | IO[Any]
 _Auth: TypeAlias = Union[tuple[str, str], _auth.AuthBase, Callable[[PreparedRequest], PreparedRequest]]
 _Cert: TypeAlias = Union[str, tuple[str, str]]
-_Files: TypeAlias = (
-    Mapping[str, SupportsRead[str | bytes] | str | bytes]
-    | Mapping[str, tuple[str | None, SupportsRead[str | bytes] | str | bytes]]
-    | Mapping[str, tuple[str | None, SupportsRead[str | bytes] | str | bytes, str]]
-    | Mapping[str, tuple[str | None, SupportsRead[str | bytes] | str | bytes, str, _TextMapping]]
-)
+# Files can be passed as either a mapping, or a list of tuples.
+_FileContent: TypeAlias = SupportsRead[str | bytes] | str | bytes
+_FileName: TypeAlias = str | None
+_FileContentType: TypeAlias = str
+_FileCustomHeaders: TypeAlias = _TextMapping
```

Comment:
`_TextMapping` is `MutableMapping[str, str]`. I think we should just delete it, because it's confusing, but in a separate PR. But for these headers:

- Does it really have to be mutable?
- Does it accept bytes keys or values?

Reply:
Thanks; I saw there was already some discussion around this here: https://github.com/python/typeshed/issues/7724#issuecomment-1115767799

The outcome so far appears to be:
- It's _probably_ not required to be mutable for `headers`, however
- It's the same type as is used elsewhere for `proxies`, which _is_ mutated, and
- The upstream maintainers aren't committed to ensuring immutability for those arguments, and ask for a `dict` in the official documentation

So I definitely agree that should be a separate PR :grin: because it sounds like a thorny issue all round.

Reply:
Bytes seem to work for me. I tried `requests.post('http://localhost:12345/', files={"x": ("foo", "bar", "text/plain", {b'x': b'y'})})`, it does create a request and probably shouldn't be an error.

Reply:
Just to be clear - is this something you'd like to be addressed in this PR? I would prefer to open that can of worms in a follow up if possible, as this PR is addressing an orthogonal issue.

Happy to open that PR after this one is closed.

Reply:
We should allow bytes and immutable mappings (in this PR), because it works at runtime. Note that even the immutable `Mapping` is invariant in the key type, so you will need `Mapping[str | Foo] | Mapping[bytes | Foo] | Mapping[str | bytes, Foo]`.

Reply:
So after some investigation, I disagree that the headers type should be changed to include `bytes`. This is also why I'd like to avoid rolling the headers type change into this PR.

With the example request you posted above, I see an invalid body prepared for the request:

```python
from requests import Request

prepared = Request(
    "POST",
    "http://localhost:12345/",
    files={"x": ("foo", "bar", "text/plain", {b"x": b"y"})},
).prepare()
print(prepared.body.decode())
```
Output:
```
--ae71473b6c428c3d94bfbd5415f7bf1e
Content-Disposition: form-data; name="x"; filename="foo"
Content-Type: text/plain
b'x': b'y'

bar
--ae71473b6c428c3d94bfbd5415f7bf1e--
```

Note the 'header' of `b'x': b'y'`, which definitely doesn't look like the header a user would expect to be sending.

This is caused by formatting inside the `urllib3` library, which results in the python representation of a bytestring:
- https://github.com/psf/requests/blob/0f358eabe6f8e4c472a5030d9cd513965868bd6c/requests/models.py#L197
- https://github.com/urllib3/urllib3/blob/6de3330eb54f73a57c7860f75123bde8b043dbd2/src/urllib3/fields.py#L244